### PR TITLE
Improve Java model generation

### PR DESCRIPTION
**Pull request by jpsim**
This PR aims to update Java model generation to handle primary, indexed & required fields. Note that I have absolutely no idea what I'm doing here, so I'll need someone from Realm Java to let me know if this is ok, or more likely what I need to fix.

Depends on #120. See https://github.com/realm/realm-browser-osx/compare/jp-generate-swift-models...jp-improve-java-model-generation for just the changes in this PR.

/cc @kneth @cmelchior @TimOliver

### CODE REVIEW

File: RealmBrowser/Classes/RLMModelExporter.m [hunk 1]
```diff
@@ -93,69 +102,80 @@ +(void)saveModels:(NSArray *)models toFolder:(NSURL *)url
...
         }
-        [model appendFormat:@"\n"];
 
-        [model appendFormat:@"public class %@ extends RealmObject {\n", schema.className];
+        [model appendFormat:@"\npublic class %@ extends RealmObject {\n", schema.className];
 
         // fields
         for (RLMProperty *property in schema.properties) {
+            if (property.isPrimary) {
+                [model appendString:@"    @PrimaryKey\n"];
```
Comment (cmelchior) [inline RealmBrowser/Classes/RLMModelExporter.m]: All these annotations also needs to be imported as well. They have the package names `io.realm.annotations.PrimaryKey`, `io.realm.annotations.Index` and `io.realm.annotations.Required`. It is legal to import them if not used, but if you want to be a really good Android citizen it should only classes it actually uses :)

**Comment by cmelchior** [comment]
Apart from the annotations also needs to be imported it looks correct from what I can tell.

File: RealmBrowser/Classes/RLMModelExporter.m (hunk 1 above)
Comment (jpsim) [inline RealmBrowser/Classes/RLMModelExporter.m]: Ok, and what should be the order of imports? Is it ok to just import all the necessary things in alphabetical order?

Comment (cmelchior) [inline RealmBrowser/Classes/RLMModelExporter.m]: Yes that is fine.  Any order will work, but most IDE's does it alphabetically.

Comment (jpsim) [inline RealmBrowser/Classes/RLMModelExporter.m]: Ah, I ended up sorting first with `RealmObject`, then annotations in the order they're added, then objects in the order they're added. Is that ok or should I just sort all imports equally?

Comment (jpsim) [inline RealmBrowser/Classes/RLMModelExporter.m]: This commit sorts them alphabetically, let me know if you prefer that: c900d82

**Comment by kneth** [comment]
Looks fine to me.

**Pull request merged**